### ISSUE TITLE
allow user to select a model on a controller to deploy to

### ISSUE THREAD

**Issue by battlemidget** [conjure-up/conjure-up#753]
For new deployments we can whitelist the 'controller' model in a spell to allow the user to deploy applications to the controller's 'bootstrap' model. Also give the user the ability to deploy to a new model (which is default behavior) or any other listed models.

This is similar to the #624 however strictly for new deployments.

**Comment by johnsca** [comment]
For the TUI, you can already specify a model name, but existing models are rejected.  That should be an easy fix.

For the GUI, I envision a screen just after controller selection that prompts for the new model name, with the current generated value as the default, and also lists any existing models below that for selection, typically filtering out the controller model unless the spell whitelists it.

That jive with your understanding?

**Comment by mikemccracken** [comment]
Aside from allowing deploys onto the controller model, what's the use case for supporting deploying into an existing model? Is this for the 'next deploy' idea?

I wonder if it's necessary at all given the "real soon now" availability of cross-model relations?

I guess for 'next deploy' you'd still need to be able to have the user pick an existing model, if we deploy into an existing model they'd be picking that, and if we deploy into a new model and cross-model-relate, you still need to pick the other model to relate into.

However I'm not sure we need this to always be there, it seems like extra unnecessary UI for the case where you're doing a first deploy. Thoughts?

**Comment by johnsca** [comment]
While this definitely ties in to the "next deploy" functionality, I think it's entirely reasonable to want to deploy a second bundle / spell into an existing model so that those can then be connected together.  It also surfaces the fact that a model is being created a bit better, so that the user has an idea of why, after they deployed two different things, they can only see the most recent.

**Comment by petevg** [comment]
@mikemccracken I'm currently working on getting matrix (https://github.com/juju-solutions/matrix) to use conjure-up in headless mode to do deploys. Unfortunately, the model thing is a blocker -- matrix really wants to make and manage a model itself, and it would be very nice if conjure-up was able to re-use that model.

**Comment by johnsca** [comment]
@petevg You would only actually need the TUI portion of this, so perhaps that could be done separately?

**Comment by petevg** [comment]
@johnsca Yep. I just need to headless part of it. If people are okay with headless behaving differently, I can submit a PR ...

**Comment by battlemidget** [comment]
@petevg @johnsca are we good on this bug? can it be closed?

**Comment by johnsca** [comment]
@battlemidget #760 only addressed the headless side of this.  We still presumably want to provide a GUI screen for influencing the generated model name or selecting an existing model to deploy in to.

Also, I wonder if we should in fact exclude the controller model by default, or just show it since the user was already presented with the option to use JaaS vs self-hosting the controller?

**Comment by battlemidget** [comment]
So if we present the controller model by default that will take care of Landscape's need to install its bits on an existing system (the bootstrap node) and not use up an additional machine just for landscape-server
